Branch-SDK/BNCDeviceInfo.m

```diff
@@ -146,6 +146,7 @@ - (NSString *)userAgentString {
 
 // IDFA should never be cached
 - (void)checkAdvertisingIdentifier {
+    self.optedInStatus = [BNCSystemObserver attOptedInStatus];
```

Comment:
So the idea is this provokes the prompt? Does it block until the user responds?

Reply:
It never provokes the prompt.  It does check what the status is though, assuming the client didn't exclude the code via Linker flag.

Reply:
Mostly cause the user could have changed it via menu instead of via prompt. So it's easier to just check it in a neutral way when we check for idfa.